docs/recipes/WritingTests.md

```diff
@@ -365,60 +344,37 @@ Consider the following `App` component:
 ```js
 import { connect } from 'react-redux'
 
-class App extends Component {
-  /* ... */
+const App = props => {
+  return <div>{props.user}</div>
 }
 
 export default connect(mapStateToProps)(App)
 ```
 
-In a unit test, you would normally import the `App` component like this:
-
-```js
-import App from './App'
-```
-
-However, when you import it, you're actually holding the wrapper component returned by `connect()`, and not the `App` component itself. If you want to test its interaction with Redux, this is good news: you can wrap it in a [`<Provider>`](https://react-redux.js.org/api/provider) with a store created specifically for this unit test. But sometimes you want to test just the rendering of the component, without a Redux store.
-
-In order to be able to test the App component itself without having to deal with the decorator, we recommend you to also export the undecorated component:
+To test it, we can build a wrapper function on top of React Testing Library's `render` function that will create a store for each test like:
 
 ```js
-import { connect } from 'react-redux'
-
-// Use named export for unconnected component (for tests)
-export class App extends Component {
-  /* ... */
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { createStore } from 'redux'
+import { Provider } from 'react-redux'
+import App from '../../containers/App'
+import { initialState, reducer } from './reducer.js'
+
+function renderWithRedux(
+  ui,
+  { initialState, store = createStore(reducer, initialState) } = {}
+) {
+  return render(<Provider store={store}>{ui}</Provider>)
 }
```

Comment:
Could we recommend the use of the `wrapper` option as well as suggest people follow the instructions in the setup docs here so they don't have to worry about the redux provider in any of their tests? https://testing-library.com/docs/react-testing-library/setup

Reply:
I was wondering about that, what will be the best way to give it an `initialState` in my test? The only option I could think of is adding an `initialState` as one of the options and pass it to the `AllTheProviders` component.. WDYT? 
Btw, I can also update the Recipe for React-Redux in RTL's docs if you'd like, since I used it to write my example here and it's also outdated.
https://testing-library.com/docs/example-react-redux

Reply:
I was wondering about that, what will be the best way to give it an `initialState` in my test? The only option I could think of is adding an `initialState` as one of the options and pass it to `AllTheProviders` component.. WDYT? 
Btw, I can also update the Recipe for React-Redux in RTL's docs if you'd like, since I used it to write my example here and it's also outdated.
https://testing-library.com/docs/example-react-redux

Reply:
The `renderWithRedux` you have here is basically the exact same thing you would have as your custom render. So you'd still be able to provide an additional argument. Here's a better example:

```javascript
function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({children}) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
}
```

Reply:
Got it, fixed it :)
Thanks for that!
I can update RTL's docs accordingly to contain the example you wrote here for the React-Redux recipe if you think we should do it.

Reply:
That would be super. Thank you!